pkg/lib/serverTime.js

```diff
@@ -439,63 +602,142 @@
         event.preventDefault();
         return false;
     }
 
     function remove_server(event, index) {
-        custom_ntp.servers.splice(index, 1);
-        change("custom_ntp", custom_ntp);
+        servers.splice(index, 1);
+        change();
         event.stopPropagation();
         event.preventDefault();
         return false;
     }
 
     function change_server(event, index, value) {
-        custom_ntp.servers[index] = value;
-        change("custom_ntp", custom_ntp);
+        servers[index] = value;
+        change();
         event.stopPropagation();
         event.preventDefault();
         return false;
     }
 
-    const ntp_servers = (
+    if (servers.length == 0)
+        servers.push("");
+
+    const server_rows = (
         <table>
             <tbody>
-                { custom_ntp.servers.map((s, i) => (
+                { servers.map((s, i) => (
                     <tr key={i}>
                         <td style={{ width: "100%" }}>
                             <input type="text" className="form-control" value={s} placeholder={_("NTP server")}
                 onChange={event => change_server(event, i, event.target.value)} />
                         </td>
                         <td>
-                            <Button variant="secondary" onClick={event => add_server(event, i)} icon={ <PlusIcon /> } />
+                            <Button variant="secondary" onClick={event => add_server(event, i)}
+                                    icon={ <PlusIcon /> } className="add-server" />
                         </td>
                         <td>
                             <Button variant="secondary" onClick={event => remove_server(event, i)}
-                                    icon={ <CloseIcon /> }
-                                    isDisabled={i == custom_ntp.servers.length - 1} />
+                                    icon={ <CloseIcon /> } className="remove-server"
+                                    isDisabled={servers.length == 1 && servers[0] == ""} />
                         </td>
                     </tr>))
                 }
             </tbody>
         </table>);
 
+    return (
+        <Validated errors={errors} error_key="ntp_servers">
+            <div id="systime-ntp-servers-row">
+                <div id="systime-ntp-servers">
+                    { server_rows }
+                </div>
+            </div>
+        </Validated>);
+};
+
+const StaticServerList = ({ servers, change, map }) => {
+    function label(name) {
+        if (map && map[name])
+            return map[name];
+        else
+            return name;
+    }
+    return <>
+        { servers.map(s => <Checkbox key={s.name}
+                                            id={"cb-" + s.name}
+                                            label={label(s.name)}
+                                            isChecked={s.enabled}
+                                            onChange={val => { s.enabled = val; change() }} />)
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230119-090841-c16a56a9-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)